### PR TITLE
added lowercase restriction

### PR DESCRIPTION
**Pull request by evnxprk**
let me know if these changes work for you. thank you 👍🏻

### CODE REVIEW

File: apps/api/src/prisma/prisma.service.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export type AuditLogEntry = {
 };
 
 export class PrismaService extends PrismaClient implements OnModuleInit {
+    user: any;
+    accessLog: any;
```
Comment (robere2) [inline apps/api/src/prisma/prisma.service.ts:18]: These are already present in `PrismaClient`, they shouldn't be necessary

File: apps/api/src/types/user/dto/update-user.input.ts [hunk 2]
```diff
@@ -5,4 +5,7 @@ import { InputType, PartialType } from "@nestjs/graphql";
  * Input type for updateUser mutation. Null values are not updated. To update a non-null value to null, explicitly pass null.
  */
 @InputType()
-export class UpdateUserInput extends PartialType(CreateUserInput) {}
+export class UpdateUserInput extends PartialType(CreateUserInput) {
+    username: any;
+    password: any;
```
Comment (robere2) [inline apps/api/src/types/user/dto/update-user.input.ts:10]: These are already present in `CreateUserInput`, they shouldn't be necessary

**Review by robere2 — CHANGES_REQUESTED** [review]
Thanks! I left a couple comments. If you added these because you were getting TypeScript errors, it could be because you don't have our Prisma types generated (our READMEs are a bit outdated 🙈 working on updating them now actually)

In case you don't have the full dev environment set up, here's some brief instructions on how to do so, at least for your purposes:

---

1. Fork/clone the repository
2. Unfortunately, our UI icons require a pro key, so you'll need to go into `apps/ui/package.json` and remove all of the `@fortawesome/*` packages temporarily. I'll be working on a fix for this.
3. Run `npm i` in the root directory
4. Add the following contents to `/apps/api/.env.docker`:
```yml
DATABASE_URL="postgresql://postgres:password@db:5432/postgres?schema=public&connect_timeout=300"
POSTGRES_PASSWORD=password

REDIS_URL=redis://redis:6379
RABBITMQ_URL=amqp://rabbitmq:5672
```
5. Start up the Postgres DB, Redis, and RabbitMQ by running `docker compose -d` in the root directory
6. Add the following contents to `/packages/prisma/.env`:
```yml
DATABASE_URL="postgresql://postgres:password@localhost:5432/postgres?schema=public&connect_timeout=300"
```
7. Run `npx prisma migrate dev` in `/packages/prisma` to initiate the database
8. Run `npm run dev` in `/apps/cli`, followed by the command `g` to generate the default groups and `u` to create your user account.
9. Add the following to `/apps/api/.env`:
```yml
NODE_ENV=development

DATABASE_URL=postgresql://postgres:password@localhost:5432/postgres?schema=public&connect_timeout=300
POSTGRES_PASSWORD=password

REDIS_URL=redis://localhost:6379
RABBITMQ_URL=amqp://localhost:5672

SESSION_SECRET=UJA0jUwEKpe7EFB9Vm6narS9bzihWYbg9f9hT0fiKGJfgwA4mUz9FLLkhJYCFZ62

DISCORD_CLIENT_ID=0
DISCORD_CLIENT_SECRET=0
DISCORD_CALLBACK_URL=https://discord.com/api/unused
DISCORD_WEBHOOK=https://discord.com/api/unused

LOCAL_IMAGE_URL=https://localhost/img/

TRUST_PROXY=loopback
HTTPS=false
```
10. Now you can run `npm run start:dev` in `/apps/api`, and access the GraphQL playground at http://localhost:4000/graphql

--- 

Once you're up and running, these queries will help you in this particular case:
```gql
mutation {
  loginLocal(username: "yourName", password: "yourPassword") {
    id
  }
}
```
This will log you into the account you created with the CLI. You'll automatically be an admin.
```gql
mutation {
  createUser(input: {
    username: "CapsName",
    mail: "example@example.com"
  }) {
    id
  }
}
```
This will create a new user with capital letters in their name
```gql
query {
  findManyUser {
    username
    id
  }
}
```
This will list all the users (well, the first 20) and usernames.

---

I'm not sure this is necessary though, unless you want to test your changes yourself. Your changes look good so once you make my requested changes I can go ahead and just confirm that it's all working as expected.

**Comment by evnxprk** [comment]
Okay, I will work on these sometime tonight or tomorrow. 

I initially went to the README and there was no information like the one you gave me, so I just went off of what I knew and what VSC changes gave me. Now that I do have things to go off of, I will work on these!

**Comment by robere2** [comment]
I didn't think about it until now, but `toLowerCase()` should also be added to the CLI:
https://github.com/rpitv/glimpse/blob/master/apps/cli/src/index.ts#L191

Don't hesitate to ask if you run into any issues or have any questions.

**Comment by robere2** [comment]
Superseded by #73

**Pull request closed without merging**